oak_restricted_kernel/src/lib.rs

```diff
@@ -156,6 +157,45 @@ pub fn start_kernel(info: &BootParams) -> ! {
     // Safety: in the linker script we specify that the ELF header should be placed at 0x200000.
     let program_headers = unsafe { elf::get_phdrs(VirtAddr::new(0x20_0000)) };
 
+    let _stage0_dice_data = {
+        let dice_memory_slice = {
+            let e820_dice_data_entry = info
+                .e820_table()
+                .iter()
+                .find(|e| e.entry_type() == Some(oak_linux_boot_params::E820EntryType::DiceData))
+                .expect("failed to find dice data");
+
+            let start_addr = {
+                let phys_addr = PhysAddr::new(
+                    e820_dice_data_entry
+                        .addr()
+                        .try_into()
+                        .expect("couldn't convert usize to u64"),
+                );
+                VirtAddr::new(phys_addr.as_u64() + mm::KERNEL_OFFSET)
```

Comment:
(a) At this point I think we're still running under identity mapping, so PhysAddr == VirtAddr.
(b) After we've loaded the proper page tables, VirtAddr = PhysAddr + mm:DIRECT_MAPPING_OFFSET.

(See Conrad's comment, below.)